### PR TITLE
De-pickle serial_task

### PR DESCRIPTION
**Pull request by dannyroberts**
## Technical Summary

https://docs.google.com/spreadsheets/d/1RykGc-ARbQtjtYv6Xch25vm4ft5Z13nc_4eewOwdtSk/edit#gid=0

This PR is best reviewed commit by commit.

There are two related changes in this PR:
- It makes `serial_task` not use `pickle` by default, and makes each place that use it explicitly pass in `serializer='pickle'`
- For the `serial_task` tasks that obviously did not need to use pickle, it switches them to using the default (json) instead.

### Safety story

The changes here are very rote and narrow, but the ability of tests to catch any human error on my part suffers because of how celery behaves differently in tests (`CELERY_TASK_ALWAYS_EAGER`) than it does in production. For that reason I'm assigning this medium risk.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/preindex/tasks.py [hunk 1]
```diff
@@ -14,7 +14,8 @@ def run_continuous_indexing_task():
     preindex_couch_views.delay()
 
 
-@serial_task('couch-continuous-indexing', timeout=60 * 60, queue=settings.CELERY_PERIODIC_QUEUE, max_retries=0)
+@serial_task('couch-continuous-indexing', timeout=60 * 60, queue=settings.CELERY_PERIODIC_QUEUE, max_retries=0,
+             serializer='pickle')
```
Comment (dannyroberts) [inline corehq/preindex/tasks.py:18]: This one is obviously not necessary. Looks like I just missed it. I'll fix.

**Review by gherceg — APPROVED** [review]
Looks great! Thanks for the easy to follow commits.

File: corehq/apps/locations/tasks.py [hunk 2]
```diff
@@ -25,7 +25,7 @@
 
 
 @serial_task("{location_type.domain}-{location_type.pk}",
-             default_retry_delay=30, max_retries=3)
+             default_retry_delay=30, max_retries=3, serializer='pickle')
 def sync_administrative_status(location_type):
```
Comment (dannyroberts) [inline corehq/apps/locations/tasks.py:29]: I'm pretty sure `location_type` is a `LocationType` object (django model), so this interface will have to be rewritten more substantially.

File: corehq/apps/data_interfaces/tasks.py [hunk 3]
```diff
@@ -157,6 +157,7 @@ def run_case_update_rules_for_domain(domain, now=None):
     timeout=36 * 60 * 60,
     max_retries=0,
     queue='case_rule_queue',
+    serializer='pickle',
 )
 def run_case_update_rules_for_domain_and_db(domain, now, run_id, case_type, db=None):
```
Comment (dannyroberts) [inline corehq/apps/data_interfaces/tasks.py:162]: `now` is a datetime, and I wasn't confident in how we're handling those, so I decided not to try to fix this one.

**Review by minhaminha — APPROVED** [review]

**Pull request merged**